src/platform/OpenThread/GenericThreadStackManagerImpl_OpenThread.cpp

```diff
@@ -1237,6 +1244,234 @@ CHIP_ERROR GenericThreadStackManagerImpl_OpenThread<ImplClass>::_SetupSrpHost(co
     return error;
 }
 
+#if CHIP_DEVICE_CONFIG_ENABLE_THREAD_DNS_CLIENT
+template <class ImplClass>
+void GenericThreadStackManagerImpl_OpenThread<ImplClass>::FromOtDnsResponseToMdnsData(otDnsServiceInfo & serviceInfo,
+                                                                                      const char * serviceType,
+                                                                                      chip::Mdns::MdnsService & mdnsService,
+                                                                                      DnsServiceTxtEntries & serviceTxtEntries)
+{
+    char protocol[chip::Mdns::kMdnsProtocolTextMaxSize + 1];
+
+    // Extract from the <hostname>.<domain-name>. the <hostname> part.
+    size_t substringSize = strlen(serviceInfo.mHostNameBuffer) - strlen(strchr(serviceInfo.mHostNameBuffer, '.'));
+    strncpy(mdnsService.mHostName, serviceInfo.mHostNameBuffer, substringSize);
+    // Append string terminating character.
+    memset(mdnsService.mHostName + substringSize, '\0', 1);
+
+    // Extract from the <type>.<protocol>.<domain-name>. the <type> part.
+    substringSize = strlen(serviceType) - strlen(strchr(serviceType, '.'));
+    strncpy(mdnsService.mType, serviceType, substringSize);
+    // Append string terminating character.
+    memset(mdnsService.mType + substringSize, '\0', 1);
+
+    // Extract from the <type>.<protocol>.<domain-name>. the <protocol> part.
+    strncpy(protocol, serviceType + substringSize + 1, chip::Mdns::kMdnsProtocolTextMaxSize);
```

Comment:
Why does `kMdnsProtocolTextMaxSize` make sense here?  The protocol might be shorter than that, based on the naming.

Reply:
Fair point, I refactored to find the next `.` character after protocol name and calculate actual size, not use the max one.